pkg/api/snapshots.go

```diff
@@ -16,7 +17,34 @@ type SnapshotResponse struct {
 
 type ListSnapshotByDateRequest struct {
 	RepositoryUUIDS []string `json:"repository_uuids"` // Repository UUIDs to find snapshots for
-	Date            string   `json:"date"`             // Exact date to search by.
+	Date            Date     `json:"date"`             // Exact date to search by.
+}
+
+type Date time.Time
+
+func (d Date) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Time(d).Format(time.RFC3339))
```

Comment:
I don't think the marshal function should force a certain format. I think it should just use whatever format was being used for date.

Reply:
Actually that doesn't make sense. Ignore me.